streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -149,7 +149,7 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
         // explicitly disable logging for global stores
         materialized.withLoggingDisabled();
         final StoreBuilder storeBuilder = new KeyValueStoreMaterializer<>(materialized).materialize();
-        final String sourceName = newProcessorName(KStreamImpl.SOURCE_NAME);
+        final String sourceName = newProcessorName(KTableImpl.SOURCE_NAME);
```

Comment:
@guozhangwang @bbejeck @vvcephei Discovered this by chance -- not sure if fixing this would result in compatibility issues? Thoughts?

Reply:
Good catch.

If we are considering topic / store names only, then no. I'd suggest we just fix it right away.

Reply:
Hmm. I'm tracing through the code, but it's certainly safer just to mark it with a `//TODO` and leave it alone.

Reply:
Yeah, I think the only concern would be if the source node name were used for the store, and I don't think it is.

Reply:
Okay, let's change it here then.

Reply:
Thanks for verifying that this is a safe change/fix.